lib/cgo/coin.block.go

```diff
@@ -0,0 +1,419 @@
+package main
+
+import (
+	"errors"
+	"reflect"
+	"unsafe"
+
+	cipher "github.com/skycoin/skycoin/src/cipher"
+	coin "github.com/skycoin/skycoin/src/coin"
+)
+
+/*
+
+  #include <string.h>
+  #include <stdlib.h>
+
+  #include "skytypes.h"
+	#include "feecalc.h"
+*/
+import "C"
+
+//export SKY_coin_NewBlock
+func SKY_coin_NewBlock(_b C.Block__Handle, _currentTime uint64, _hash *C.cipher__SHA256, _txns C.Transactions__Handle, pFeeCalc *C.FeeCalculator, _arg2 *C.Block__Handle) (____error_code uint32) {
+	feeCalc := func(pTx *coin.Transaction) (uint64, error) {
+		var fee C.GoUint64_
+		handle := registerTransactionHandle(pTx)
+		result := C.callFeeCalculator(pFeeCalc, handle, &fee)
+		closeHandle(Handle(handle))
+		if result == SKY_OK {
+			return uint64(fee), nil
+		} else {
+			return 0, errors.New("Error calculating fee")
```

Comment:
Should work as follows :

- if error code is well known (like those included in [known error list](https://github.com/simelo/skycoin/blob/olemis_t1629_libskycoin_errors/lib/cgo/libsky_error.go#L317-L427) ) then translate it into the corresponding error object.
- else an instance of a (not yet implemented) custom API error type should be returned , and this instance should include an error code field

Reply:
error translation needs merge from `olemis_t1629_libskycoin_errors` branch .